pkg/tenant/projects.go

```diff
@@ -221,6 +221,52 @@ func (s *Server) TenantProjectCreate(c *gin.Context) {
 	c.JSON(http.StatusCreated, tproject.ToAPI())
 }
 
+// TenantProjectPatch applies a partial update to a project identified by the tenantID
+// and projectID in the URL.
+//
+// Route: /tenant/:tenantID/projects/:projectID
+func (s *Server) TenantProjectPatch(c *gin.Context) {
+	var (
+		err                        error
+		orgID, tenantID, projectID ulid.ULID
+	)
+
+	// orgID is required to check ownership of the project
+	if orgID = orgIDFromContext(c); ulids.IsZero(orgID) {
+		return
+	}
+
+	// Parse the tenantID from the URL
+	tenantParam := c.Param("tenantID")
+	if tenantID, err = ulid.Parse(tenantParam); err != nil {
+		sentry.Warn(c).Err(err).Str("tenantID", tenantParam).Msg("could not parse tenant id from URL")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("tenant not found"))
+		return
+	}
+
+	// Parse the projectID from the URL
+	projectParam := c.Param("projectID")
+	if projectID, err = ulid.Parse(projectParam); err != nil {
+		sentry.Warn(c).Err(err).Str("projectID", projectParam).Msg("could not parse project id from URL")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("project not found"))
+		return
+	}
+
+	// Verify that the tenant exists in the user's organization.
+	if err = db.VerifyOrg(c, orgID, tenantID); err != nil {
+		if errors.Is(err, db.ErrNotFound) || errors.Is(err, db.ErrOrgNotVerified) {
+			c.JSON(http.StatusNotFound, api.ErrorResponse("tenant not found"))
+			return
+		}
+		sentry.Warn(c).Err(err).Msg("could not check verification")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("could not verify organization"))
```

Comment:
I think this should be a 500 error not a 401 -- most cases are captured by the not found above -- so this is probably a database error.